azurerm/internal/services/storage/client.go

```diff
@@ -0,0 +1,84 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2019-04-01/storage"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/authorizers"
+	"github.com/tombuildsstuff/giovanni/storage/2018-11-09/file/directories"
+)
+
+type Client struct {
+	// this is currently unexported since we only use it to look up the account key
+	// we could export/use this in the future - but there's no point it being public
+	// until that time
+	accountsClient storage.AccountsClient
+}
+
+// NOTE: this temporarily diverges from the other clients until we move this client in here
+// once we have this, can take an Options like everything else
+func BuildClient(accountsClient storage.AccountsClient) *Client {
+	return &Client{
+		accountsClient: accountsClient,
+	}
+}
+
+func (client Client) FindResourceGroup(ctx context.Context, accountName string) (*string, error) {
+	accounts, err := client.accountsClient.List(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("Error listing Storage Accounts (to find Resource Group for %q): %s", accountName, err)
+	}
+
+	if accounts.Value == nil {
+		return nil, nil
```

Comment:
I was torn on this - if we get an invalid api response for this feels like the right thing to do, but this would block `terraform refresh` from detecting that the resource doesn't exist should the API return an invalid response; WDYT?

Reply:
My concern is that the error message from the resource call using an empty `accounts.Value` could be confusing. Do you know what error message is returned when this happens? Does the api error and say that the resource group was empty/not found?

Reply:
we assume it's gone - so we'll `d.SetID("")` and mark the resource as 'gone' - since if it's not there there's not a lot we can do